internal/vercheck/vercheck.go

```diff
@@ -15,32 +15,60 @@ import (
 
 	"github.com/fatih/color"
 	"github.com/pkg/errors"
-	"golang.org/x/mod/semver"
-
+	"github.com/samber/lo"
 	"go.jetpack.io/devbox/internal/boxcli/usererr"
 	"go.jetpack.io/devbox/internal/build"
 	"go.jetpack.io/devbox/internal/cmdutil"
 	"go.jetpack.io/devbox/internal/envir"
 	"go.jetpack.io/devbox/internal/ux"
 	"go.jetpack.io/devbox/internal/xdg"
+	"golang.org/x/mod/semver"
 )
 
 // Keep this in-sync with latest version in launch.sh.
 // If this version is newer than the version in launch.sh, we'll print a notice.
 const expectedLauncherVersion = "v0.2.0"
 
+// versionCheckEnvName determines whether the version check has already occurred.
+// We set this env-var so that this devbox command invoking other devbox commands
+// do not re-run the version check and print the notice again.
+const versionCheckEnvName = "DEVBOX_VERSION_CHECK"
+
 // currentDevboxVersion is the version of the devbox CLI binary that is currently running.
 // We use this variable so that we can mock it in tests.
 var currentDevboxVersion = build.Version
 
+// isDevBuild determines whether this CLI binary was built during development, or published
+// as a release.
+// We use this variable so we can mock it in tests.
+var isDevBuild = build.IsDev
```

Comment:
One thing we could do is make all the build values unexported (I think ld flags work with unexported values) and expose everything via a public struct. That way we can just mock the struct.